src/components/Table/GridTable.tsx

```diff
@@ -284,11 +292,26 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     return rows;
   }, [columns, rows, sorting, sortState]);
 
-  const columnSizes = useMemo(
-    () => calcColumnSizes(columns, style.nestedCards?.firstLastColumnWidth),
-    [columns, style.nestedCards?.firstLastColumnWidth],
+  const [columnSizes, setColumnSizes] = useState<string[]>(
+    calcColumnSizes(columns, style.nestedCards?.firstLastColumnWidth, undefined, style.minWidth),
   );
 
+  const [tableWidth, setTableWidth] = useState<number | undefined>();
+
+  const debouncedTableWidth = useDebouncedCallback((width) => {
+    setTableWidth(width);
+    setColumnSizes(calcColumnSizes(columns, style.nestedCards?.firstLastColumnWidth, width, style.minWidth));
+  }, 100);
```

Comment:
Chose 100, as it feels pretty snappy. People taking longer than 100ms to resize their screen may cause multiple re-renders. I'd rather wait to see if that is an issue before introducing more lag.

Reply:
Yeah, seems like 1-re-render-every-100ms while they're dragging things around is really pretty reasonable. Definitely a lot better than 1-re-render-every-pixel-move-of-the-mouse. :-)